File: app/src/main/java/ru/schultetabledima/schultetable/advice/AdvicePresenter.java
package ru.schultetabledima.schultetable.advice;

import android.util.Log;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

import moxy.InjectViewState;
import moxy.MvpPresenter;
import ru.schultetabledima.schultetable.R;
import ru.schultetabledima.schultetable.contracts.AdviceContract;

@InjectViewState
public class AdvicePresenter extends MvpPresenter<AdviceContract.View> implements AdviceContract.Presenter {

    private final AdviceContract.Model adviceModel;
    private List<Integer> adviceResource;


    public AdvicePresenter(AdviceModel adviceModel) {
        this.adviceModel = adviceModel;
        init();
        pushAdviceToView();
    }


    private void init() {
        adviceResource = new ArrayList<>(3);

        if (Locale.getDefault().getLanguage().equals("ru")) {
            adviceResource.add(R.raw.advice1_ru);
            adviceResource.add(R.raw.advice2_ru);
            adviceResource.add(R.raw.advice3_ru);
        } else {
            adviceResource.add(R.raw.advice1_en);
            adviceResource.add(R.raw.advice2_en);
            adviceResource.add(R.raw.advice3_en);
        }
    }

//    public void setModel(AdviceModel adviceModel) {
//        this.adviceModel = adviceModel;
//    }

    private void pushAdviceToView() {
        int count = 0;
        for (int id : adviceResource) {
            getViewState().showAdvice(count++, adviceModel.getAdvice(id));
        }
    }
}

File: app/src/main/java/ru/schultetabledima/schultetable/advice/AdvicePresenter.kt
package ru.schultetabledima.schultetable.advice

import moxy.InjectViewState
import moxy.MvpPresenter
<<<<<<< HEAD
import ru.schultetabledima.schultetable.contracts.AdviceContract

@InjectViewState
class AdvicePresenter(private val adviceModel: AdviceContract.Model) :
    MvpPresenter<AdviceContract.View?>(), AdviceContract.Presenter {

    private var listAdvice: List<String>? = null

    init {
=======
import ru.schultetabledima.schultetable.R
import ru.schultetabledima.schultetable.contracts.AdviceContract
import java.util.*

@InjectViewState
class AdvicePresenter : MvpPresenter<AdviceContract.View?>(), AdviceContract.Presenter {


    private var adviceModel: AdviceContract.Model? = null
    private var adviceResource: Int? = null
    private var listAdvice: List<String>? = null

    init {
        initialization()
>>>>>>> devdesign
        getAdvice()
        pushAdviceToView()
    }

<<<<<<< HEAD
    private fun getAdvice() {
        listAdvice = adviceModel.advice
=======
    private fun initialization() {
        adviceModel = AdviceModel()
        adviceResource =
            if (Locale.getDefault().language == "ru") R.raw.advice1_ru else R.raw.advice1_en
    }

    private fun getAdvice() {
        listAdvice = adviceModel!!.getAdvice(adviceResource!!).lines()
>>>>>>> devdesign
    }

    private fun pushAdviceToView() {
        for ((count, advice) in listAdvice!!.withIndex()) {
            viewState!!.showAdvice(count, advice)
        }
    }
}